### PR TITLE
Fix loading state in DiscoverSheet

### PR DESCRIPTION
**Pull request by osdnk**
Fixes RNBW-2215

This fixes a few issues:

1. The ghost is cut-off, this is just a CSS issue. This reverts #2137 partially, but I tested and the original issue is gone anyway
2. While loading, the ghost should not be visible, because the content might still appear. We show the ghost only when loading is done/failed.
~3. While fetching a new query with having some results already (e.g., when I inputted `USD` and now I add `C` to `USDC` we should keep the old results until new results will appear. Otherwise, we see an empty state for a short while, which is weird~

## PoW (screenshots / screen recordings)
BEFORE:
https://streamable.com/7jioo0

AFTER:
https://streamable.com/6j3jgo

## Final checklist
[ ] Assigned individual reviewers?
[ ] Added labels?

### CODE REVIEW

**Review by derHowie — APPROVED** [review]
👍

**Review by skylarbarrera — APPROVED** [review]
🔥

**Review by derHowie — APPROVED** [review]

**Review by skylarbarrera — APPROVED** [review]
👌

**Comment by jinchung** [comment]
When typing in a query with no results, then typing character by character after that initial query, it flashes the previous result before going back to no result, back and forth: https://recordit.co/nTy8n4foam

**Review by jinchung — CHANGES_REQUESTED** [review]
Just wanted to move this into "request changes" so that Skylar can add a few tweaks so others don't merge quite yet!

File: src/components/exchange/CurrencySelectionList.js [hunk 1]
```diff
@@ -62,10 +62,12 @@ const CurrencySelectionList = (
       testID={testID}
       transition={skeletonTransition}
     >
-      {showList && (
+      {showList && !showSkeleton && (
         <Centered flex={1}>
-          {showNoResults ? (
-            <NoCurrencyResults />
+          {showNoResults && !loading ? (
+            loading ? null : (
```
Comment (jinchung) [inline src/components/exchange/CurrencySelectionList.js:68]: it will never reach line 68 bc `!loading` is specified on line 67

Comment (jinchung) [reply src/components/exchange/CurrencySelectionList.js:68]: @osdnk

**Review by jinchung — COMMENTED** [review]
On the very first time when I open search and search for a token that has no results (e.g, "Vowevadsf"), it just shows the loading shimmer forever. If I delete it and then type it again, THEN it shows me the ghost.

**Comment by jinchung** [comment]
Mentioned to Skylar: I had merged his PR for ENS handling, but as of now when I search for "jackson" and see the ENS results, then I replace it with some gibberish, the ENS results still hang out

**Comment by osdnk** [comment]
Fixes https://linear.app/rainbow/issue/RNBW-2252/serach-bar-is-not-fixed-while-searching-on-android

**Review by jinchung — APPROVED** [review]
🌮

**Comment by skylarbarrera** [comment]
send it

**Pull request merged**